plugins/processors/noise/noise.go

```diff
@@ -0,0 +1,136 @@
+package noise
+
+import (
+	"math"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+const (
+	defaultSensitivity = 1.0
+	defaultEpsilon     = 1.0
+)
+
+var sampleConfig = `
+  [[processors.noise]]
+    sensitivity = 1.0
+    epsilon = 1.0
+    ignore_fields = []
+    ignore_measurements = []
+`
+
+var laplaceValue = 0.0
+var fieldExcludeSet = make(map[string]bool)
+var measurementExcludeSet = make(map[string]bool)
+
+type Noise struct {
+	Sensitivity        float64
+	Epsilon            float64
+	IgnoreFields       []string        `toml:"ignore_fields"`
+	IgnoreMeasurements []string        `toml:"ignore_measurements"`
+	Log                telegraf.Logger `toml:"-"`
+}
+
+func (p *Noise) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Noise) Description() string {
+	return "Generates noise based on the Laplace distribution and add that to " +
+		"all numerical field values. Exclusions can be listed in the config file."
+}
+
+// Gets a random, positive Integer from a Laplace distribution and adds it to a
+// given value
+func (p *Noise) addLaplaceUInt64(value uint64) uint64 {
+	return value + value*uint64(math.Round(math.Abs(p.getRandomLaplaceNoise())))
```

Comment:
I don't think you should use the `Abs` of the noise. This way you can infer that the actual value was always smaller than the observed value... How about
```suggestion
	v := float(value) + p.getRandomLaplaceNoise()
	if v < 0 {
		return 0
	}
	return uint64(math.Round(v))
```
You might even want to leave-out the rounding

Reply:
I agree, Abs() is not really helpful here. I'll check that when gaussian & uniform distribution support is done.

Reply:
Implemented with da8adb88d10e9a1d5d9d278971a37a2032dfd843